### PR TITLE
Update example, fix null-pointer check, update android build

### PR DESCRIPTION
**Pull request by mikehardy**
Hi there - thought you might like the work I did to polish this module up so I could use it in my app

The commit messages have the details but the basics are:

- android build system using new versions and updated ext version definition style
- fix type error in the sms list null pointer check recently committed #22 
- refresh the example completely for react-native 0.58.6, incl. RN59 + API28 local bundle workaround for debug builds
- integrate permissions check in example so it works out of the box
- remove WRITE_SMS from example since Android disallows this for apps that aren't the default SMS app in Android >= KitKat

All of this makes the example current and usable as a demonstrator, with the code working correctly on my API28 / Android Pie/9 device. 

Thanks!

### CODE REVIEW

**Comment by briankabiro** [comment]
Thanks for the contribution, @mikehardy!

I appreciate this a lot. The tests will come in handy - I wasn't sure how to set them up but this will go a long way in ensuring confidence in the code.

File: example/package.json [hunk 1]
```diff
@@ -1,23 +1,24 @@
...
+  "version": "0.0.1",
+  "private": true,
+  "scripts": {
+    "start": "node node_modules/react-native/local-cli/cli.js start",
+    "test": "jest"
+  },
+  "dependencies": {
+    "react": "16.6.3",
+    "react-native": "0.58.6",
+    "react-native-get-sms-android": "github:mikehardy/react-native-get-sms-android"
```
Comment (briankabiro) [inline example/package.json]: Think we can update this to point to the main repo. I've merged the PR already but we can spin up another one.

Comment (mikehardy) [reply example/package.json]: Yes! That was an oversight on my part. By way of excuse, it is hard to fix the main module and the example at the same time without making this mistake - I noticed it was pointing to some other random repo to start, I pointed it to main in one commit then (while doing some final testing) pointed it back to me and forgot it :-)

Comment (briankabiro) [reply example/package.json]: No worries at all.

**Pull request merged**